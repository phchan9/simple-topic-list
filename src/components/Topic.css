/* make ordered-list decimal-indexed */
.Topic {
    list-style: decimal inside;
    display: list-item;
    border: 0 none;
}
