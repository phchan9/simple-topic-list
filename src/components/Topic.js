import React from 'react';
import PropTypes from 'prop-types';

// make ordered-list decimal-indexed
const topicStyle = {
  listStyle: 'decimal inside',
  display: 'list-item',
  border: '0 none',
};

/* eslint-disable react/jsx-filename-extension */
const Topic = ({ title, upVotes, downVotes, onUpVoteCallBack, onDownVoteCallBack }) => (
  <li style={topicStyle} className="list-group-item">
    <h5>{title}</h5>
    <div>
      <button
        type="button"
        className="btn btn-default"
        onClick={onUpVoteCallBack}
      >
        UpVote <span className="badge">{upVotes}</span>
      </button>
      <button
        type="button"
        className="btn btn-default"
        onClick={onDownVoteCallBack}
      >
        DownVote <span className="badge">{downVotes}</span>
      </button>
    </div>
  </li>
);

Topic.propTypes = {
  title: PropTypes.string.isRequired,
  upVotes: PropTypes.number.isRequired,
  downVotes: PropTypes.number.isRequired,
  onUpVoteCallBack: PropTypes.func.isRequired,
  onDownVoteCallBack: PropTypes.func.isRequired,
};

export default Topic;
