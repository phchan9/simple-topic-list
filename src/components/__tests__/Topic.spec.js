import React from 'react';
import { shallow } from 'enzyme';
import Topic from '../Topic';

const setup = () => {
  const props = {
    title: 'This is the title',
    upVotes: 30,
    downVotes: 50,
    onUpVoteCallBack: jest.fn(),
    onDownVoteCallBack: jest.fn()
  };

  const wrapper = shallow(<Topic {...props  }/>);

  return {
    props,
    wrapper
  };
};

describe('Topic Component', () => {
  it('render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should properly fire `onUpVotecallback` or `onDownVoteCallBack` when click button'
     , () => {
    const { wrapper, props } = setup();
    const buttons = wrapper.find('.btn-default');
    const upVoteButtonWrapper = buttons.at(0);
    const downVoteButtonWrapper = buttons.at(1);

    upVoteButtonWrapper.simulate('click');
    expect(props.onUpVoteCallBack).toHaveBeenCalled();
    downVoteButtonWrapper.simulate('click');
    expect(props.onDownVoteCallBack).toHaveBeenCalled();

  });
});




// behavior called
