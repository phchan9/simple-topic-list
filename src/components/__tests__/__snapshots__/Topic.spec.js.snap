// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Component render correctly 1`] = `
<li
  className="list-group-item"
  style={
    Object {
      "border": "0 none",
      "display": "list-item",
      "listStyle": "decimal inside",
    }
  }
>
  <h5>
    This is the title
  </h5>
  <div>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="button"
    >
      UpVote 
      <span
        className="badge"
      >
        30
      </span>
    </button>
    <button
      className="btn btn-default"
      onClick={[Function]}
      type="button"
    >
      DownVote 
      <span
        className="badge"
      >
        50
      </span>
    </button>
  </div>
</li>
`;
